woocommerce/woocommerce-android
# Save shipping methods locally

This PR depends on [this FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/3013)

Closes: #11515

### Description
This PR adds support for caching shipping methods and improves the perceived performance when displaying shipping methods. With these changes, the source of truth will be the local data, and on background, we will refresh the shipping methods info. I also added support for refreshing the info using the pull-to-refresh pattern.

### Testing instructions

TC1
1. Open the app
2. Go to orders
3. Tap on add new order (+)
4. Add a product to enable the the shipping section
5. Tap on Add shipping
6. Tap on Select method to navigate to the select method screen
7. Check that if is the first time that the screen is opened the data is fetched and displayed form a network request
8. Check that the n time the screen is opened the data is displayed from the DB (on background we make a request to refresh the data)

TC2
1. Additionally you can test the refresh scenario

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/18119390/9ac62a5a-5b6f-40eb-8e29-2e000c5109aa
